### ISSUE TITLE
Unable to render parent component from button within a facet

### ISSUE THREAD

**Issue by skuzzle** [TheCoder4eu/BootsFaces-OSP#381]
(I'm using BootsFaces Version 0.8.5, PrimeFaces 5.3 and both running on a web stack built with Google Guice and Apache my-faces 2.2.8)

I'm encountering the following problem of which I'm pretty sure it is a bug. I have a composite component which is inside a `f:facet` to have it inserted as footer in a dialog. Its used like this (the dialog is an abstraction of Primeface's dialog built by me). The `okCancel` component is supposed to render the dialog's buttons. The OK Button has to re-render the dialog contents in case that there is a validation error. The `update` clause is directly passed through to the Ok button.

```
        <h:form id="addAccountForm" class="with-dirty-check">
            <d:dialog 
                    id="addAccountDialog"
                    title="Create new account"
                    dialogName="addAccDialog"
                    onClose="#{addAccountDialog.onClose}"
                    width="400">

                <f:facet name="footer">
                    <d:okCancel id="addAccountOkCancel" update=":addAccountForm:addAccountDialog" 
okAction="#{addAccountController.addAccount}"/>
                </f:facet>
            </d:dialog>
        </h:form>
```

The implementation of the okCancel component looks like this:

```
    <composite:implementation>
        <div class="clearfix">
            <h:outputText value="#{cc.parent.clientId}:dialog"></h:outputText>
            <b:button onclick="closeDialog();" value="#{cc.attrs.cancelText}" id="cancel"/>
            <b:commandButton styleClass="pull-right" 
                    id="ok"
                    action="#{cc.attrs.okAction}" 
                    icon="#{cc.attrs.okIcon}" 
                    update="#{cc.attrs.update}"
                    look="success" value="#{cc.attrs.okText}">
            </b:commandButton>
        </div>
    </composite:implementation>
```

Now whatever ID I pass to the update clause causes an error to be rendered. No matter whether it is an absolute or relative ID. I can't even render a child of the same naming container within the facet. This is what the DOM looks like and the rendered error message:

![html](https://cloud.githubusercontent.com/assets/317179/15337467/402b88b8-1c7c-11e6-9487-3f8ea3e9f3d0.PNG)

The actual stack trace is burried somewhre in the off of the HTML response. This is the trace I obtained from eclipse while debugging the behavior:

```
Daemon Thread [http-bio-8080-exec-6] (Suspended (breakpoint at line 45 in ExpressionResolver))  
    owns: GuiceFilter$Context  (id=467) 
    owns: SocketWrapper<E>  (id=468)    
    ExpressionResolver.getComponentIDs(FacesContext, UIComponent, String) line: 45  
    AJAXRenderer.generateAJAXCallForClientBehavior(FacesContext, IAJAXComponent, ClientBehavior) line: 453  
    AJAXRenderer.generateBootsFacesAJAXAndJavaScript(FacesContext, ClientBehaviorHolder, ResponseWriter, String, String, boolean) line: 177 
    AJAXRenderer.generateBootsFacesAJAXAndJavaScript(FacesContext, ClientBehaviorHolder, ResponseWriter) line: 154  
    CommandButtonRenderer.encodeBegin(FacesContext, UIComponent) line: 97   
    CommandButton(UIComponentBase).encodeBegin(FacesContext) line: 596  
    CommandButton(UIComponentBase).encodeAll(FacesContext) line: 526    
    UIPanel(UIComponentBase).encodeAll(FacesContext) line: 550  
    HtmlCompositeComponentRenderer.encodeChildren(FacesContext, UIComponent) line: 66   
    UINamingContainer(UIComponentBase).encodeChildren(FacesContext) line: 643   
    UINamingContainer(UIComponentBase).encodeAll(FacesContext) line: 541    
    HtmlCompositeFacetRenderer.encodeChildren(FacesContext, UIComponent) line: 80   
    UIOutput(UIComponentBase).encodeChildren(FacesContext) line: 643    
    UIOutput(UIComponentBase).encodeAll(FacesContext) line: 541 
    DialogRenderer.encodeFooter(FacesContext, Dialog) line: 170 
    DialogRenderer.encodeMarkup(FacesContext, Dialog) line: 117 
    DialogRenderer.encodeEnd(FacesContext, UIComponent) line: 47    
    Dialog(UIComponentBase).encodeEnd(FacesContext) line: 674   
    Dialog(UIComponentBase).encodeAll(FacesContext) line: 554   
    UIPanel(UIComponentBase).encodeAll(FacesContext) line: 550  
    HtmlCompositeComponentRenderer.encodeChildren(FacesContext, UIComponent) line: 66   
    UINamingContainer(UIComponentBase).encodeChildren(FacesContext) line: 643   
    UINamingContainer(UIComponentBase).encodeAll(FacesContext) line: 541    
    HtmlForm(UIComponentBase).encodeAll(FacesContext) line: 550 
    Container(UIComponentBase).encodeAll(FacesContext) line: 550    
    HtmlBody(UIComponentBase).encodeAll(FacesContext) line: 550 
    UIViewRoot(UIComponentBase).encodeAll(FacesContext) line: 550   
    FaceletViewDeclarationLanguage.renderView(FacesContext, UIViewRoot) line: 1891  
    ViewHandlerImpl.renderView(FacesContext, UIViewRoot) line: 313  
    RestorableViewHandler(ViewHandlerWrapper).renderView(FacesContext, UIViewRoot) line: 58 
    RenderResponseExecutor.execute(FacesContext) line: 116  
    LifecycleImpl.render(FacesContext) line: 267    
    FacesServlet.service(ServletRequest, ServletResponse) line: 200 
    FacesHttpServlet.service(ServletRequest, ServletResponse) line: 51  
    ServletDefinition.doServiceImpl(HttpServletRequest, HttpServletResponse) line: 287  
    ServletDefinition.doService(ServletRequest, ServletResponse) line: 277  
    ServletDefinition.service(ServletRequest, ServletResponse) line: 182    
    ManagedServletPipeline.service(ServletRequest, ServletResponse) line: 91    
    FilterChainInvocation.doFilter(ServletRequest, ServletResponse) line: 85    
    CustomPersistFilter.doFilter(ServletRequest, ServletResponse, FilterChain) line: 53 
    FilterChainInvocation.doFilter(ServletRequest, ServletResponse) line: 82    
    SimpleFilterChain.doFilter(ServletRequest, ServletResponse) line: 44    
    FormAuthenticationFilter(AdviceFilter).executeChain(ServletRequest, ServletResponse, FilterChain) line: 108 
    FormAuthenticationFilter(AdviceFilter).doFilterInternal(ServletRequest, ServletResponse, FilterChain) line: 137 
    FormAuthenticationFilter(OncePerRequestFilter).doFilter(ServletRequest, ServletResponse, FilterChain) line: 125 
    SimpleFilterChain.doFilter(ServletRequest, ServletResponse) line: 41    
    GuiceShiroFilter(AbstractShiroFilter).executeChain(ServletRequest, ServletResponse, FilterChain) line: 449  
    AbstractShiroFilter$1.call() line: 365  
    SubjectCallable<V>.doCall(Callable<V>) line: 90 
    SubjectCallable<V>.call() line: 83  
    WebDelegatingSubject(DelegatingSubject).execute(Callable<V>) line: 383  
    GuiceShiroFilter(AbstractShiroFilter).doFilterInternal(ServletRequest, ServletResponse, FilterChain) line: 362  
    GuiceShiroFilter(OncePerRequestFilter).doFilter(ServletRequest, ServletResponse, FilterChain) line: 125 
    FilterChainInvocation.doFilter(ServletRequest, ServletResponse) line: 82    
    ManagedFilterPipeline.dispatch(ServletRequest, ServletResponse, FilterChain) line: 119  
    GuiceFilter$1.call() line: 133  
    GuiceFilter$1.call() line: 130  
    GuiceFilter$Context.call(Callable<T>) line: 203 
    GuiceFilter.doFilter(ServletRequest, ServletResponse, FilterChain) line: 130    
    ApplicationFilterChain.internalDoFilter(ServletRequest, ServletResponse) line: 241  
    ApplicationFilterChain.doFilter(ServletRequest, ServletResponse) line: 208  
    StandardWrapperValve.invoke(Request, Response) line: 220    
    StandardContextValve.invoke(Request, Response) line: 122    
    NonLoginAuthenticator(AuthenticatorBase).invoke(Request, Response) line: 501    
    StandardHostValve.invoke(Request, Response) line: 170   
    ErrorReportValve.invoke(Request, Response) line: 98 
    AccessLogValve.invoke(Request, Response) line: 950  
    StandardEngineValve.invoke(Request, Response) line: 116 
    CoyoteAdapter.service(Request, Response) line: 408  
    Http11Processor(AbstractHttp11Processor<S>).process(SocketWrapper<S>) line: 1040    
    Http11Protocol$Http11ConnectionHandler(AbstractProtocol$AbstractConnectionHandler<S,P>).process(SocketWrapper<S>, SocketStatus) line: 607   
    JIoEndpoint$SocketProcessor.run() line: 313 
    ThreadPoolExecutor(ThreadPoolExecutor).runWorker(ThreadPoolExecutor$Worker) line: 1142  
    ThreadPoolExecutor$Worker.run() line: 617   
    TaskThread(Thread).run() line: 744  
```

I'm stuck on the problem of rendering  the dialog content for like 2 days now. I hope you guys have some advice of what is going on here or at least are able to reproduce and fix the bug in a future version

**Comment by stephanrauh** [comment]
Did you already try one of our advanced search expressions? The simplest one being `@form`, or `**:addAccountDialog`, or something like this. Thing is, the custom component adds another `NamingContainer` to the JSF tree, and that often causes confusion. The advanced search expressions should solve the problem.

**Comment by stephanrauh** [comment]
We've solved this bug recently, as you can see at https://showcase.bootsfaces.net/miscellaneous/CompositeComponents.jsf. As far as I remember, it was solved in 1.0.2 or 1.0.1.
